controllers/chat/moveToSign.js

```diff
@@ -84,12 +84,16 @@ const moveToSign = async (req, res) => {
       })
     );
 
+    const oldChannelName = createdChannel?.channel?.name?.trim()?.split(',');
+
     try {
-      if (!newChannel?.data?.name) {
+      if (!oldChannelName) {
+        const newChannelName = [userModel?.username, targetUserModel?.username].join(', ');
+        createdChannel.channel.name = newChannelName;
         await newChannel.updatePartial({
           set: {
             channel_type: isContainAnonimous ? CHANNEL_TYPE.ANONYMOUS : CHANNEL_TYPE.CHAT,
-            name: [userModel?.username, targetUserModel?.username].join(', '),
+            name: newChannelName,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic here seems to be correct, but there's a potential issue with the `oldChannelName` check. If `createdChannel?.channel?.name` is an empty string, `oldChannelName` will also be an empty string, which is a falsy value in JavaScript. This means that the new channel name will be set even if the old channel name was intentionally set to an empty string. To avoid this, you should explicitly check if `oldChannelName` is `undefined` or `null`.

```diff
-      if (!oldChannelName) {
+      if (oldChannelName === undefined || oldChannelName === null) {
         const newChannelName = [userModel?.username, targetUserModel?.username].join(', ');
         createdChannel.channel.name = newChannelName;
         await newChannel.updatePartial({
           set: {
             channel_type: isContainAnonimous ? CHANNEL_TYPE.ANONYMOUS : CHANNEL_TYPE.CHAT,
             name: newChannelName,
             better_channel_member: newStateMemberWithAnonInfo
           }
         });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->